### ISSUE TITLE
[java] std_set.i not supported yet

### ISSUE THREAD

**Issue by albertfc** [swig/swig#931]
$ cat test.i 
%module "std-set-test"
%include <std_set.i>

**$ swig -c++ -java -o test.cpp -I/usr/share/swig3.0/java -I/usr/share/swig3.0/std test.i 
/usr/share/swig3.0/std/std_set.i:98: Error: Syntax error in input(3).**

$ swig -version
SWIG Version 3.0.13

Compiled with g++ [x86_64-pc-linux-gnu]

Configured options: +pcre

**Comment by ojwb** [comment]
You shouldn't be adding `-I` options for paths in the SWIG standard library (or need to) - the bug here seems to be that `%include <std_set.i>` doesn't work for Java, but does for at least some other languages - e.g. Ruby and Python:

```
$ swig -c++ -java -o test.cpp test.i
test.i:2: Error: Unable to find 'std_set.i'
$ swig -c++ -ruby -o test.cpp test.i
$ swig -c++ -python -o test.cpp test.i
$
```

**Comment by ojwb** [comment]
Looks like `std_set.i` is currently only available for Ruby, Python and Scilab:

```
$ git ls-files '*/std_set.i'
Lib/python/std_set.i
Lib/ruby/std_set.i
Lib/scilab/std_set.i
Lib/std/std_set.i
```

If you want to add support, it's probably just a matter of adapting one of the other language-specific `std_set.i` files for Java (I'd also look at `Lib/java/std_*.i` for tips).

**Comment by ojwb** [comment]
There was a ticket in the old SF tracker about this, which I've now closed as a duplicate: https://sourceforge.net/p/swig/patches/253/

However, amongst some dead URLs there was https://chadretz.wordpress.com/2009/11/27/stl-collections-with-java-and-swig/ which might be useful for someone wanting to implement this.

**Comment by malaterre** [comment]
Here is what I have been using so far:

* https://github.com/malaterre/GDCM/blob/master/Wrapping/Java/std_set.i

**Comment by ojwb** [comment]
This was fixed in 2018 by 0f18b430fb39d44e126f9f981462c47fca7b8dba
